signer/src/transaction_signer.rs

```diff
@@ -762,21 +752,27 @@ where
         Ok(())
     }
 
-    #[tracing::instrument(skip(self))]
-    async fn get_signer_public_keys(
-        &mut self,
-        bitcoin_chain_tip: &model::BitcoinBlockHash,
+    /// Get the set of public keys for the current signing set.
+    ///
+    /// If there is a successful `rotate-keys` transaction in the database
+    /// then we should use that as the source of truth for the current
+    /// signing set, otherwise we fall back to the bootstrap keys in our
+    /// config.
+    #[tracing::instrument(skip_all)]
+    pub async fn get_signer_public_keys(
+        &self,
+        chain_tip: &model::BitcoinBlockHash,
     ) -> Result<BTreeSet<PublicKey>, Error> {
-        let last_key_rotation = self
-            .context
-            .get_storage()
-            .get_last_key_rotation(bitcoin_chain_tip)
-            .await?
-            .ok_or(Error::MissingKeyRotation)?;
-
-        let signer_set = last_key_rotation.signer_set.into_iter().collect();
-
-        Ok(signer_set)
+        let db = self.context.get_storage();
+
+        // Get the last rotate-keys transaction from the database on the
+        // canonical Stacks blockchain (which we identify using the
+        // canonical bitcoin blockchain). If we don't have such a
+        // transaction then get the bootstrap keys from our config.
+        match db.get_last_key_rotation(chain_tip).await? {
+            Some(last_key) => Ok(last_key.signer_set.into_iter().collect()),
+            None => Ok(self.context.config().signer.bootstrap_signing_set()),
```

Comment:
Should we have an intermediate fallback here to the DKG shares, like we do in tx coordinator?

Reply:
Good catch, yes we need to update that query to use the `dkg_share` table if the `rotate_keys_transactions` table is empty.

Reply:
Fixed in https://github.com/stacks-network/sbtc/pull/655/commits/7d8ae77e7f816e9e949ea8e95c611c68af19afbb.